### PR TITLE
ts: Remove SPL coders from Anchor package

### PR DESCRIPTION
**Pull request by acheroncrypto**
Follow up on #2143, coders in Anchor package is no longer necessary. This is a breaking change, codebases that rely on SPL from Anchor package would need to migrate to the new packages and only import as needed.

### CODE REVIEW

File: ts/packages/anchor/src/program/accounts-resolver.ts [hunk 1]
```diff
@@ -232,31 +230,17 @@ export class AccountsResolver<IDL extends Idl, I extends AllInstructions<IDL>> {
...
     const address = publicKey.toString();
     if (!this._cache.has(address)) {
-      if (name === "TokenAccount") {
-        const accountInfo = await this._provider.connection.getAccountInfo(
-          publicKey
-        );
-        if (accountInfo === null) {
-          throw new Error(`invalid account info for ${address}`);
-        }
-        const data = coder().accounts.decode("token", accountInfo.data);
```
Comment (acheroncrypto) [inline ts/packages/anchor/src/program/accounts-resolver.ts:254]: Could just decode token accounts without any coder, not sure if it's needed though.

**Comment by Henry-E** [comment]
Not if sure @tomlinton or @callensm is best to check this out

File: ts/packages/anchor/src/program/accounts-resolver.ts (hunk 1 above)
Comment (callensm) [reply ts/packages/anchor/src/program/accounts-resolver.ts:254]: if this is going to be removed, would be worth also investigating where and why `TokenAccounts` are being added to the account store cache and managing that too.

Comment (callensm) [reply ts/packages/anchor/src/program/accounts-resolver.ts:254]: maybe @armaniferrante has insight there

**Comment by callensm** [comment]
it's my understanding that the spl coders here are mildly broken anyway, so i wouldn't care if they were removed in favor of the official spl packages that have their own encoding/decoding, but would almost certain be a breaking change for many.

**Comment by callensm** [comment]
> Cool, thanks for that. A breaking change doesn't sound great for sure. Might be a close then unfortunately.

i'd still consider merging once we get more input of the usability of these coders. if the impact is minimal might be worth getting rid of the bloat /shrug

File: ts/packages/anchor/src/program/accounts-resolver.ts (hunk 1 above)
Comment (acheroncrypto) [reply ts/packages/anchor/src/program/accounts-resolver.ts:254]: Completely removing this caused a problem that I don't remember so I added a coder for only what we need and nothing more in https://github.com/coral-xyz/anchor/pull/2155/commits/57f0f7e660ce3a559fa6b225ad2c772de5a7db13

**Comment by acheroncrypto** [comment]
> Cool, thanks for that. A breaking change doesn't sound great for sure. Might be a close then unfortunately.

We had a talk about this with @armaniferrante and his input was to completely remove them and allow people to import as needed.

**Comment by callensm** [comment]
ok, just verified. we're good to merge this

**Comment by Henry-E** [comment]
Great, @acheroncrypto if you wouldn't mind to fix the merge conflicts when you get a chance and will try merge ASAP!

**Comment by acheroncrypto** [comment]
Fixed the conflicts, only `zero-copy` test seems to fail and it's not related to this PR.

**Comment by Henry-E** [comment]
@acheroncrypto are you happy with merging or any last bits you want to add?

**Pull request merged**